src/test/scala-2/chiselTests/testing/scalatest/TestingDirectorySpec.scala

```diff
@@ -90,4 +96,21 @@ class TestingDirectorySpec extends AnyFunSpec with Matchers with SimulatorAPI wi
 
   }
 
+  describe("A test that sets a subdirectory") {
+
+    it("should have a subdirectory") {
+      seed.withValue(Some("foo")) {
+        checkDirectoryStructure(
+          "build",
+          "chiselsim",
+          "TestingDirectorySpec",
+          "A-test-that-sets-a-subdirectory",
+          "should-have-a-subdirectory",
+          "foo"
+        )
+      }
```

Comment:
I'm a little iffy on this being done via a DynamicVariable rather than by changing the `HasTestingDirectory`. What I'm proposing works a lot better in Scala 3 with Context Functions, but works in Scala 2 as well, a couple of ways it can work:

Provide a new implicit val:
```suggestion
      // Scala 2
      implicit val testDir: HasTestingDirectory = implicitly[HasTestingDirectory].withSeed("foo")
      // Scala 3
      given HasTestingDirectory = summon[HasTestingDirectory].withSeed("foo")
      checkDirectoryStructure(
        "build",
        "chiselsim",
        "TestingDirectorySpec",
        "A-test-that-sets-a-subdirectory",
        "should-have-a-subdirectory",
        "foo"
      )
```

Or with context functions (and Scala 2's crappy version):

```suggestion
      // Scala 2
      testingDirectoryWithSeed(Some("foo")) { implicit testDir =>
      // Scala 3
      testingDirectoryWithSeed(Some("foo")) {
        checkDirectoryStructure(
          "build",
          "chiselsim",
          "TestingDirectorySpec",
          "A-test-that-sets-a-subdirectory",
          "should-have-a-subdirectory",
          "foo"
        )
      }
```

Reply:
I'm not opposed your approach here and it doesn't preclude changing to the approach I'm proposing in the future (especially in the Brave New World of Scala 3), just something to consider.